servicemodels/e2sm_ccc/servicemodel/servicemodel.go

```diff
@@ -0,0 +1,388 @@
+		ranconfigurationStructure := &topoapi.RanconfigurationStructure{
+			Name: string(supportedConfigurationStructures.GetRanConfigurationStructureName().GetValue()),
+		}
+		var attributes []*topoapi.Attribute
+		for _, supportedAttribute := range supportedConfigurationStructures.GetListOfSupportedAttributes().GetValue() {
+			ricServices := &topoapi.RICServices{}
+			var eventTriggers []*topoapi.CCCEventTriggerStyle
+			for _, ricService := range supportedAttribute.GetSupportedServices().GetEventTrigger().GetListOfSupportedEventTriggerStyles().GetValue() {
+				eventTriggers = append(eventTriggers, &topoapi.CCCEventTriggerStyle{
+					Name: ricService.GetEventTriggerStyleName().Value,
+					Type: ricService.GetEventTriggerStyleType().Value,
+					FormatType: ricService.GetEventTriggerFormatType().Value,
+				})
+				ricServices.EventTriggerStyles = eventTriggers
+			}
+
+			var controlStyles []*topoapi.CCCControlStyle
+			for _, ricService := range supportedAttribute.GetSupportedServices().GetControlService().GetListOfSupportedControlStyles().GetValue() {
+				controlStyles = append(controlStyles, &topoapi.CCCControlStyle{
+					Name: ricService.GetControlServiceStyleName().Value,
+					Type: ricService.GetControlServiceStyleType().Value,
+					ControlHeaderFormatType: ricService.GetControlServiceHeaderFormatType().Value,
+					ControlMessageFormatType: ricService.GetControlServiceMessageFormatType().Value,
+					RicProcessIdFormatType: ricService.GetRicCallProcessIdformatType().Value,
+					ControlOutcomeFormatType: ricService.GetControlServiceControlOutcomeFormatType().Value,
+				})
+				ricServices.ControlStyles = controlStyles
+			}
+
+			var reportStyles []*topoapi.CCCReportStyle
+			for _, ricService := range supportedAttribute.GetSupportedServices().GetReportService().GetListOfSupportedReportStyles().GetValue() {
+				reportStyles = append(reportStyles, &topoapi.CCCReportStyle{
+					Name: ricService.GetReportServiceStyleName().Value,
+					Type: ricService.GetReportServiceStyleType().Value,
+					// For now not decoding these elements
+					// EventTriggerStyleForReport: ,
+					ActionDefinitionFormatType: ricService.GetReportServiceActionDefinitionFormatType().Value,
+  					IndicationHeaderFormatType: ricService.GetReportServiceIndicationHeaderFormatType().Value,
+  					IndicationMessageFormatType: ricService.GetReportServiceIndicationMessageFormatType().Value,
+				})
+				ricServices.ReportStyles = reportStyles
+			}
+
+			attributes = append(attributes, &topoapi.Attribute{
+				Name: string(supportedAttribute.GetAttributeName().GetValue()),
+				RicServices: ricServices,
+			})
+		}
+
+		log.Debugf("attributes: %v", attributes)
+		ranconfigurationStructure.Attribute = attributes
+		ranFunction.RanStructures = append(ranFunction.RanStructures, ranconfigurationStructure)
+	}
+
+	log.Debugf("ranFunction.RanStructures: %v", ranFunction.RanStructures)
+
+	// TODO: Extract ListOfCellsForRANFunctionDefinition
+	// ...
```

Comment:
Will that be addressed?

Reply:
No, for now the implementation includes only the `Node-level` stuff. We did not do anything related to the `Cell-level` stuff. So, we can leave the "TODO" for future implementation of everything related to the `Cell-level`